File: tests/fixtures/ruby_repo/lib/demo/service.rb
require_relative "base"
require_relative "support"
require "set"

module Demo
  class Service < Base
    include Logging
    extend FactoryMethods

    def initialize
      @helper = Helper.new
    end

    def run(value = 1, &block)
      worker = Helper.new
      worker.save(value)
      @helper.save(value)
      yield(value) if block_given?
      block.call(value) if block
      value
    end

    def self.build(value)
      new(value)
    end
  end
end

File: tests/fixtures/php_repo/src/Service.php
<?php

namespace Demo\Service;

use Demo\Support\{Helper as H, Logs};
use function Demo\Support\make_helper as make;

final class RunnerService extends BaseService implements RunnerContract
{
    use Logs;

    private H $helper;

    public function __construct(H $helper)
    {
        $this->helper = $helper;
    }

    public function run(int $value): H
    {
        $worker = $this->helper;
        $worker->save((string) $value);
        $this->log(make()->format((string) $value));
        return $worker;
    }

    public static function create(): self
    {
        return new self(make());
    }
}
